### ISSUE TITLE
[BUG] User

### ISSUE THREAD

**Issue by stuart909** [Pengin-Open-Source/pengin-pi-2#23]
**# Bug Report: Foreign Key Relationship Issue in Django Admin User Model

## Description

When attempting to save changes to the User model in the Django admin interface, an `IntegrityError` is raised with the message `FOREIGN KEY constraint failed`. This occurs when manually toggling columns such as `is_superuser` or `is_staff`.

## Steps to Reproduce

1. Go to the Django admin interface.
2. Navigate to the User model.
3. Select a user and attempt to edit their details.
4. Toggle `is_superuser` or `is_staff`.
5. Save the changes.

## Expected Behavior

The user details should be saved without any errors.

## Actual Behavior

An `IntegrityError` is raised with the message `FOREIGN KEY constraint failed`.

## Stack Trace

```
IntegrityError at /admin/main/user/6ae5bf85-edc9-45f7-bf84-2366fdb8ed2a/change/
FOREIGN KEY constraint failed
```

## Additional Information

- The issue seems to be related to foreign key constraints in the User model.
- We are using Django's built-in `User` model with custom foreign key relationships.
- The issue occurs only when using the Django admin interface.

## Potential Fixes

1. **Check Foreign Key Relationships:**
   Ensure that all foreign key relationships involving the `User` model are correctly defined and that there are no orphaned records in related tables.

2. **Django Admin Customization:**
   Customize the admin form for the `User` model to handle related objects properly.

3. **Verify Data Integrity:**
   Run a database check to ensure there are no orphaned records or foreign key violations.

4. **Check Signals or Constraints:**
   Review any signals or constraints that might be interfering with the save operation.

### Model Definition

```python
from django.conf import settings
from django.db import models

class ExampleModel(models.Model):
    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
    # other fields...
```

###Admin Customization
```
from django.contrib import admin
from django.contrib.auth.admin import UserAdmin as BaseUserAdmin
from django.contrib.auth.models import User

class UserAdmin(BaseUserAdmin):
    # Customize the admin form if needed
    pass

admin.site.unregister(User)
admin.site.register(User, UserAdmin)
```

###Data Integrity Check
```
from django.db import connection

def check_foreign_keys():
    with connection.cursor() as cursor:
        cursor.execute("PRAGMA foreign_key_check")
        return cursor.fetchall()

# Call this function to see if there are any issues
issues = check_foreign_keys()
if issues:
    print("Foreign key issues found:", issues)
```

**Comment by mvetom** [comment]
I don't think this is an issue on the newsetup-related branches - it isn't on forums.

Somehow,  on a branch that has this issue (the blogs branch),  the django_admin_log has a reference to the wrong user - it used the default.

`	FOREIGN KEY("user_id") REFERENCES "auth_user"("id") DEFERRABLE INITIALLY DEFERRED`

On the forums branch,  for example,  which does not have this bug - django_admin_log references main_user instead,  like we want it to.

Not sure how the django_admin_log user FK got generated this way initially,  but,  for example,  on my blogs branch,  there is some code in main's admin.py to setup a UserAdmin class, which extends the BaseUserAdmin class.  Whereas on the Forums branch (which does not have this bug),  it does not register a UserAdmin class,  only our User Class. 

Another possibility that comes to mind is that perhaps the initial creation of the django_admin_log table on the non-working branches happened before the custom user in main was specified,  and no migration code was ever run that specifically updated that particular table with a reference to our custom user
